hubblestack/extmods/hubble_mods/time_sync.py

```diff
@@ -0,0 +1,223 @@
+        AuditCheckValidationError: For any validation error
+    """
+    log.debug('Module: time_sync Start validating params for check-id: {0}'.format(block_id))
+
+    chain_args = None if not extra_args else extra_args.get('chaining_args')
+    ntp_servers = _get_ntp_servers(block_id, block_dict, chain_args)
+
+    if not ntp_servers:
+        raise HubbleCheckValidationError('No ntp_servers provided')
+
+    log.debug('Validation success for check-id: {0}'.format(block_id))
+
+def execute(block_id, block_dict, extra_args=None):
+    """
+    Function that queries a list of NTP servers and checks if the
+    offset is bigger than `max_offset` minutes. It expects the results from
+    at least `nb_servers` servers in the list, otherwise the check fails.
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': ['server1', 'server2'], 'status': True},
+                  'caller': 'Audit'}
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing stat module for id: {0}'.format(block_id))
+
+    chain_args = None if not extra_args else extra_args.get('chaining_args')
+    ntp_servers = _get_ntp_servers(block_id, block_dict, chain_args)
+    
+    time_sync_result = []
+
+    for ntp_server in ntp_servers:
+        offset = _query_ntp_server(ntp_server)
+        if not offset:
+            time_sync_result.append({
+                'ntp_server': ntp_server,
+                'replied': False
+            })
+            continue
+
+        time_sync_result.append({
+            'ntp_server': ntp_server,
+            'replied': True,
+            'offset': offset
+        })
+
+    return runner_utils.prepare_positive_result_for_module(block_id, time_sync_result)
+
+def get_filtered_params_to_log(block_id, block_dict, extra_args=None):
+    """
+    For getting params to log, in non-verbose logging
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': ['server1', 'server2'], 'status': True},
+                  'caller': 'Audit'}
+    """
+    log.debug('get_filtered_params_to_log for id: {0}'.format(block_id))
+
+    #fetch required param
+    chain_args = None if not extra_args else extra_args.get('chaining_args')
+    ntp_servers = _get_ntp_servers(block_id, block_dict, chain_args)
+    return {'ntp_servers': ntp_servers}
+
+def _get_ntp_servers(block_id, block_dict, chain_args):
+    ntp_servers = runner_utils.get_param_for_module(block_id, block_dict, 'ntp_servers')
+    ntp_servers_chained = runner_utils.get_chained_param(chain_args)
+
+    extend_chained = runner_utils.get_param_for_module(block_id, block_dict, 'extend_chained')
+    if extend_chained != False:
```

Comment:
no need of this block now. We can provide default value as well in get_param_for_module